cpp/tests/io/csv_test.cpp

```diff
@@ -2194,4 +2194,31 @@ TEST_F(CsvReaderTest, DtypesMapInvalid)
   EXPECT_THROW(cudf_io::read_csv(in_opts), cudf::logic_error);
 }
 
+TEST_F(CsvReaderTest, CsvDefaultOptionsWriteReadMatch)
+{
+  auto const filepath = temp_env->get_temp_dir() + "issue.csv";
+
+  // make up some kind of dataframe
+  auto int_column = column_wrapper<int32_t>{10, 20, 30};
+  auto str_column = column_wrapper<cudf::string_view>{"abc", "mno", "xyz"};
+  cudf::table_view input_table(std::vector<cudf::column_view>{int_column, str_column});
+
+  // write that dataframe to a csv using default options to some temporary file
+  cudf_io::csv_writer_options writer_options =
+    cudf_io::csv_writer_options::builder(cudf_io::sink_info{filepath}, input_table);
+  cudf_io::write_csv(writer_options);
+
+  // read the temp csv file using default options
+  cudf_io::csv_reader_options read_options =
+    cudf_io::csv_reader_options::builder(cudf_io::source_info{filepath})
+      .header(-1)
+      .dtypes(std::vector<data_type>{dtype<int32_t>(), dtype<cudf::string_view>()});
+
+  cudf_io::table_with_metadata new_table_and_metadata = cudf_io::read_csv(read_options);
+
+  // check to see / assert / verify they are identical, or at least as identical as expected.
+  const auto new_table_view = new_table_and_metadata.tbl->view();
+  EXPECT_EQ(input_table, new_table_view);
```

Comment:
CUDF_TEST_EXPECT_TABLES_EQUAL must be used to compare table views. This is why the compilation fails.